### PR TITLE
TGP-894 - explicit audit event for profile set up

### PR DESCRIPTION
**Pull request by ChrisCookOC**

### CODE REVIEW

File: app/factories/AuditEventFactory.scala [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+case class AuditEventFactory() {
+
+  def createSetUpProfileEvent(traderProfile: TraderProfile, startTime: Option[Instant], affinityGroup: AffinityGroup)(
+    implicit hc: HeaderCarrier
+  ): DataEvent = {
+    val auditDetails = Map(
+      "eori"          -> traderProfile.actorId,
+      "affinityGroup" -> affinityGroup.toString,
+      "ukimsNumber"   -> traderProfile.ukimsNumber,
+      "hasNirms"      -> traderProfile.nirmsNumber.isDefined.toString,
```
Comment (liandra-giacomine) [inline app/factories/AuditEventFactory.scala]: Spoke to Kara, hasNirms, hasNiphl can be removed, we only need to add norms and niphls if they're defined as opposed to using boolean params swell

Comment (ChrisCookOC) [reply app/factories/AuditEventFactory.scala]: Done :)

File: test/factories/AuditEventFactorySpec.scala [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        auditDetails("ukimsNumber") mustBe "XIUKIM47699357400020231115081800"
+        auditDetails.get("nirmsNumber") mustBe None
+        auditDetails.get("niphlNumber") mustBe None
+        auditDetails("journeyStart") mustBe start.toString
+        Instant.parse(auditDetails("journeyEnd")).isAfter(start) mustBe true
+
+      }
+
+    }
+
```
Comment (besscerule) [inline test/factories/AuditEventFactorySpec.scala:82]: Do we need to test for "should error in x circumstance" here?

File: test/services/AuditServiceSpec.scala [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+          .createSetUpProfileEvent(eqTo(traderProfile), eqTo(startTime), eqTo(AffinityGroup.Individual))(any())
+      }
+
+      withClue("Should have submitted the created event to the audit connector") {
+        verify(mockAuditConnector, times(1)).sendEvent(eqTo(fakeAuditEvent))(any(), any())
+      }
+
+    }
+
+    "must let the play error handler deal with an future failure" in {
```
Comment (besscerule) [inline test/services/AuditServiceSpec.scala:103]: ahh ignore above comment :P

**Review by besscerule — APPROVED** [review]
LGTM

**Review by Ewan-James-Donovan — APPROVED** [review]

**Pull request merged**